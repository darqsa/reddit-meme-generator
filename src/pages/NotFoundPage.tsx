import Header from "../components/Header/Header";
import Modal from "../components/Modal/Modal";

const NotFoundPage = (): JSX.Element => {
  return (
    <>
      <Header currentPage={"notFoundPage"} />
      <Modal type="error" />
    </>
  );
};
export default NotFoundPage;
