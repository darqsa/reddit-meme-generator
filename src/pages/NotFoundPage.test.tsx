import { BrowserRouter } from "react-router-dom";
import TestRenderer from "react-test-renderer";
import NotFoundPage from "./NotFoundPage";

describe("Given a NotFoundPage page", () => {
  describe("When it's rendered in the screen", () => {
    test("Then it should show the NotFoundPage page", () => {
      const expectedNotFoundPage = TestRenderer.create(
        <BrowserRouter>
          <NotFoundPage />
        </BrowserRouter>
      );

      expect(expectedNotFoundPage).toMatchSnapshot();
    });
  });
});
