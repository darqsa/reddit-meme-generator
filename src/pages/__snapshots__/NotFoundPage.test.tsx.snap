// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a NotFoundPage page When it's rendered in the screen Then it should show the NotFoundPage page 1`] = `
Array [
  <div
    className="sc-kDDrLX aOAcq header-container"
  >
    <h1
      className="sc-iqcoie iTPDGW header-container__title"
    >
      Page Not Found
    </h1>
    <a
      href="/home"
      onClick={[Function]}
    >
      <button
        className="sc-bczRLJ jLdlHz"
      >
        Return to Main Page
      </button>
    </a>
  </div>,
  <div
    className="sc-crXcEl gQIaRC modal-container"
  >
    <div
      className="modal-container__popup--error"
    >
      <img
        alt="Sad meme frog"
        className="modal-container__image"
        src="img/errorModal.png"
      />
      <span>
        Somenthing went wrong, I'm sad...
      </span>
    </div>
  </div>,
]
`;
